openstudiocore/src/model/SurfacePropertyConvectionCoefficients.hpp

```diff
@@ -0,0 +1,171 @@
+ *  works may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without
+ *  specific prior written permission from Alliance for Sustainable Energy, LLC.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER, THE UNITED STATES GOVERNMENT, OR ANY CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *  AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **********************************************************************************************************************/
+
+#ifndef MODEL_SURFACEPROPERTYCONVECTIONCOEFFICIENTS_HPP
+#define MODEL_SURFACEPROPERTYCONVECTIONCOEFFICIENTS_HPP
+
+#include "ModelAPI.hpp"
+#include "ModelObject.hpp"
+#include "Surface.hpp"
+#include "SubSurface.hpp"
+#include "InternalMass.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+class Schedule;
+
+namespace detail {
+
+  class SurfacePropertyConvectionCoefficients_Impl;
+
+} // detail
+
+/** SurfacePropertyConvectionCoefficients is a ModelObject that wraps the OpenStudio IDD object 'OS:SurfaceProperty:ConvectionCoefficients'. */
+class MODEL_API SurfacePropertyConvectionCoefficients : public ModelObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit SurfacePropertyConvectionCoefficients(const Surface& surface);
+
+  explicit SurfacePropertyConvectionCoefficients(const SubSurface& surface);
+
+  explicit SurfacePropertyConvectionCoefficients(const InternalMass& surface);
+
+  virtual ~SurfacePropertyConvectionCoefficients() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> convectionCoefficient1LocationValues();
+
+  static std::vector<std::string> convectionCoefficient1TypeValues();
+
+  static std::vector<std::string> convectionCoefficient2LocationValues();
+
+  static std::vector<std::string> convectionCoefficient2TypeValues();
+
+  /** @name Getters */
+  //@{
+
+  ModelObject surfaceAsModelObject() const;
+
+  boost::optional<Surface> surfaceAsSurface() const;
+
+  boost::optional<SubSurface> surfaceAsSubSurface() const;
+
+  boost::optional<InternalMass> surfaceAsInternalMass() const;
+
+  boost::optional<std::string> convectionCoefficient1Location() const;
+
+  boost::optional<std::string> convectionCoefficient1Type() const;
+
+  boost::optional<double> convectionCoefficient1() const;
```

Comment:
My inclination would have been to make a little more friendly interface like 'setInsideCoefficient' and 'setOutsideCoefficient' with more smarts (e.g. reject setting a schedule if it is the wrong type) but I really can't fault you for sticking closer to E+.  It's just easy to shoot yourself in the foot using this object, probably ok since it is a bit of an advanced feature.